python/lsst/meas/algorithms/objectSizeStarSelector.py

```diff
@@ -192,7 +192,10 @@ def _kcenters(yvec, nCluster, useMedian=False, widthStdAllowed=0.15):
             break
 
         for i in range(nCluster):
-            centers[i] = func(yvec[clusterId == i])
+            if (clusterId == i).any():
```

Comment:
This seems like magic without a comment explaining what is going on. The `clusterId == i` is also used later on so maybe it would look less magic if a variable was used to avoid the repetition?